v2/devdoc/log_context_property_basic_types_requirements.md

```diff
@@ -82,4 +82,339 @@ typedef LOG_CONTEXT_PROPERTY_TYPE (*LOG_CONTEXT_PROPERTY_TYPE_GET_TYPE)(void);
 
 **SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_010: [** `LOG_CONTEXT_PROPERTY_TYPE_IMPL(int64_t).get_type` shall returns the property type `LOG_CONTEXT_PROPERTY_TYPE_int64_t`. **]**
 
-Note: the rest of the types will be implemented by cloning int32_t, thus will be done in a separate PR.
+## LOG_CONTEXT_PROPERTY_TYPE_IMPL(uint64_t).to_string
+
+```c
+typedef int (*LOG_CONTEXT_PROPERTY_TYPE_TO_STRING)(void* property_value, char* buffer, size_t buffer_length);
+```
+
+`LOG_CONTEXT_PROPERTY_TYPE_IMPL(uint64_t).to_string` produces the string representation of an `uint64_t`.
+
+**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_011: [** If `property_value` is `NULL`, `LOG_CONTEXT_PROPERTY_TYPE_IMPL(uint64_t).to_string` shall fail and return -1. **]**
+
+**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_012: [** `LOG_CONTEXT_PROPERTY_TYPE_IMPL(uint64_t).to_string` shall call `snprintf` with `buffer`, `buffer_length` and format string `PRIu64` and pass in the values list the `uint64_t` value pointed to be `property_value`. **]**
+
+**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_013: [** `LOG_CONTEXT_PROPERTY_TYPE_IMPL(uint64_t).to_string` shall succeed and return the result of `snprintf`. **]**
+
+**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_014: [** If any error is encountered, `LOG_CONTEXT_PROPERTY_TYPE_IMPL(uint64_t).to_string` shall fail and return -1. **]**
```

Comment:
>return -1 [](http://example.com/codeflow?start=148&length=9)

A pedantic Nitpick: Technically this is not entirely accurate.  The function snprintf will return a negative number if it fails it doesn't say anything in the spec or the Microsoft doc about what negative number it returns.  This would only matter if the user specifically is looking for -1 and not just a value that is negative.

Reply:
After I wrote a long dissertation I realized what you want to say.
I shall change this spec to say return a negative value.